src/main/java/io/split/android/client/service/splits/SplitChangeProcessor.java

```diff
@@ -20,25 +36,62 @@ public ProcessedSplitChange process(SplitChange splitChange) {
-        return buildProcessedSplitChange(Collections.singletonList(split), changeNumber);
+    public ProcessedSplitChange process(Split featureFlag, long changeNumber) {
+        return buildProcessedSplitChange(Collections.singletonList(featureFlag), changeNumber);
     }
 
     @NonNull
-    private static ProcessedSplitChange buildProcessedSplitChange(List<Split> featureFlags, long changeNumber) {
-        List<Split> activeSplits = new ArrayList<>();
-        List<Split> archivedSplits = new ArrayList<>();
-        for (Split split : featureFlags) {
-            if (split.name == null) {
+    private ProcessedSplitChange buildProcessedSplitChange(List<Split> featureFlags, long changeNumber) {
+        List<Split> activeFeatureFlags = new ArrayList<>();
+        List<Split> archivedFeatureFlags = new ArrayList<>();
+        for (Split featureFlag : featureFlags) {
+            if (featureFlag.name == null) {
                 continue;
             }
-            if (split.status == Status.ACTIVE) {
-                activeSplits.add(split);
+
+            if (mConfiguredSets.isEmpty()) {
+                processAccordingToStatus(activeFeatureFlags, archivedFeatureFlags, featureFlag);
             } else {
-                archivedSplits.add(split);
+                processAccordingToSets(activeFeatureFlags, archivedFeatureFlags, featureFlag);
             }
         }
 
-        return new ProcessedSplitChange(activeSplits, archivedSplits, changeNumber, System.currentTimeMillis() / 100);
+        return new ProcessedSplitChange(activeFeatureFlags, archivedFeatureFlags, changeNumber, System.currentTimeMillis() / 100);
+    }
+
+    /**
+     * Process the feature flag according to its status
+     */
+    private void processAccordingToStatus(List<Split> activeFeatureFlags, List<Split> archivedFeatureFlags, Split featureFlag) {
+        if (featureFlag.status == Status.ACTIVE) {
+            activeFeatureFlags.add(featureFlag);
+        } else {
+            archivedFeatureFlags.add(featureFlag);
+        }
+    }
+
+    /**
+     * Process the feature flag according to its sets
+     */
+    private void processAccordingToSets(List<Split> activeFeatureFlags, List<Split> archivedFeatureFlags, Split featureFlag) {
+        if (featureFlag.sets == null || featureFlag.sets.isEmpty()) {
+            archivedFeatureFlags.add(featureFlag);
+            return;
+        }
+
+        boolean shouldArchive = true;
+        for (String set : featureFlag.sets) {
+            if (mConfiguredSets.contains(set)) {
+                // If the feature flag has at least one set that matches the configured sets,
+                // we process it according to its status
+                processAccordingToStatus(activeFeatureFlags, archivedFeatureFlags, featureFlag);
+                shouldArchive = false;
+                break;
+            }
+        }
+
+        if (shouldArchive) {
+            archivedFeatureFlags.add(featureFlag);
+        }
```

Comment:
I would verify: if this is a new split that does not exist in storage, and does not have sets, it will be added to archivedFeatureFlags to be deleted eventually, make sure the deletion does not raise exception if the split does not exist.